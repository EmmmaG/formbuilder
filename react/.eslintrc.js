module.exports = {
    globals: {},
    env: {
        commonjs: true,
        es6: true,
    },
    parser: '@typescript-eslint/parser',
    plugins: ['@typescript-eslint'],
    parserOptions: {
        ecmaVersion: 2018,
        ecmaFeatures: {
            globalReturn: false,
        },
        sourceType: 'module',
    },
}
