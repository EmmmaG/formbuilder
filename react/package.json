{
  "name": "hcard-builder",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@material-ui/icons": "^3.0.2",
    "@types/react-dom": "^16.8.3",
    "react": "^16.8.4",
    "react-dom": "^16.8.4",
    "react-scripts": "2.1.8",
    "typescript": "^3.3.4000"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ],
  "description": "This project was to conver a fixed format file into csv",
  "main": "index.js",
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@typescript-eslint/eslint-plugin": "^1.5.0",
    "@typescript-eslint/parser": "^1.5.0",
    "@types/enzyme": "^3.1.11",
    "@types/enzyme-adapter-react-16": "^1.0.1",
    "babel-core": "^6.26.3",
    "babel-loader": "^7.1.5",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
    "babel-plugin-transform-object-rest-spread": "^6.23.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-runtime": "^6.25.0",
    "enzyme": "^3.2.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^24.1.0",
    "jest-enzyme": "^7.0.1",
    "jest-fetch-mock": "^1.6.5",
    "ts-jest": "^22.4.6"
  },
  "author": "",
  "license": "ISC"
}
