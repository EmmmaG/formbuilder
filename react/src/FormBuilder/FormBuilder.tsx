import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import OutlinedInput from '@material-ui/core/OutlinedInput'
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import Button from '@material-ui/core/Button'

import TextInputBuilder, { TextInput } from '../Components/TextInputBuilder'
import RadioSelectBuilder, { RadioSelect } from '../Components/RadioSelectBuilder'
import CheckboxBuilder, { Checkbox } from '../Components/CheckboxBuilder'

const styles = () => ({
	select: {
		height: '35px',
		backgroundColor: 'white',
	},
	selectRoot: {
		display: 'flex',
		alignItems: 'center',
	},
})

export type Field = TextInput | RadioSelect | Checkbox
export type FieldType = 'textInput' | 'radioSelect' | 'checkbox'

export interface Props extends WithStyles<typeof styles> {
	onFieldsCreate: (field: Field) => void
	fetchData: () => Promise<void>
}

export interface State {
	fieldType: FieldType
}

export class FormBuilder extends React.Component<Props, State> {
	public constructor(props: Props) {
        super(props)

        this.state = {
            fieldType: 'textInput',
        }
	}

	private handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
		this.setState({ fieldType: event.target.value as FieldType })
	}

	private renderOptions = () => [{
		value: 'textInput',
		label: 'Text Input',
	}, {
		value: 'radioSelect',
		label: 'Radio Select',
	}, {
		value: 'checkbox',
		label: 'Checkbox',
	}].map(o => (
		<option value={o.value} key={o.value}>
			{o.label}
		</option>
	))

	private renderFieldBuilder = () => {
		const { fieldType } = this.state
        const { onFieldsCreate } = this.props

		if (fieldType === 'textInput') {
			return <TextInputBuilder onCreate={onFieldsCreate} />
		}

		if (fieldType === 'radioSelect') {
			return <RadioSelectBuilder onCreate={onFieldsCreate} />
		}

		if (fieldType === 'checkbox') {
			return <CheckboxBuilder onCreate={onFieldsCreate} />
		}
	}

	public render() {
        const { classes } = this.props

		return (
			<div>
				<Typography>Please select a field type to create</Typography>
				<FormControl variant='outlined'>
					<Select
						style={{ width: '250px' }}
						className={classes.select}
						classes={{ root: classes.selectRoot }}
						native
						value={this.state.fieldType}
						onChange={this.handleChange}
						input={
							<OutlinedInput
								name={'field type'}
								labelWidth={0}
								id={'type'}
							/>
						}
					>
						{this.renderOptions()}
					</Select>
				</FormControl>

				<Typography>Or load a existing form from server</Typography>
				<Button
					variant='outlined'
					color='primary'
					onClick={this.props.fetchData}
				>
					Load saved form
				</Button>

				<div>
					{this.renderFieldBuilder()}
				</div>
            </div>
		)
	}
}

export default withStyles(styles)(FormBuilder)
