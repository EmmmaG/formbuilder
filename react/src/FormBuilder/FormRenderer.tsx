import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import { Field } from './FormBuilder'
import { TextInput } from '../Components/TextInputBuilder'
import { RadioSelect } from '../Components/RadioSelectBuilder'
import { Checkbox } from '../Components/CheckboxBuilder'

import TextInputRenderer from '../Components/TextInputRenderer'
import RadioSelectRenderer from '../Components/RadioSelectRenderer'
import CheckboxRenderer from '../Components/CheckboxRenderer'

const styles = () => ({
    container: {
        minWidth: '400px',
        height: '100vh',
        overflowY: 'auto' as 'auto',
    },
    title: {
        fontSize: '25px',
        margin: '10px',
    },
})

export interface Props extends WithStyles<typeof styles> {
    fields: Field[]
    onDelete: (id: number) => void
}

export class FormRenderer extends React.Component<Props> {
    private onDelete = (id: number) => {
        this.props.onDelete(id)
    }

    private renderTextInput = (field: TextInput) => <TextInputRenderer key={field.id} field={field} onDelete={this.onDelete} />
    private renderRadioSelect = (field: RadioSelect) => <RadioSelectRenderer key={field.id} field={field} onDelete={this.onDelete} />
    private renderCheckbox = (field: Checkbox) => <CheckboxRenderer key={field.id} field={field} onDelete={this.onDelete} />

    private renderForms = () => {
        const { fields } = this.props

        return fields.map(field => {
            if (field.type === 'textInput') {
                return this.renderTextInput(field)
            }
            if (field.type === 'radioSelect') {
                return this.renderRadioSelect(field)
            }
            if (field.type === 'checkbox') {
                return this.renderCheckbox(field)
            }
        })
    }

	public render() {
        const { classes } = this.props

		return (
			<Paper className={classes.container}>
                <Typography className={classes.title}>Form Preview</Typography>

                {this.renderForms()}
            </Paper>
		)
	}
}

export default withStyles(styles)(FormRenderer)
