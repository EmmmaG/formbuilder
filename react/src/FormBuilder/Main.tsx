import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import FormBuilder, { Field } from './FormBuilder'
import FormRenderer from './FormRenderer'

const styles = () => ({
    container: {
        display: 'flex',
        flexDirection: 'row' as 'row',
        justifyContent: 'space-evenly',
        flexWrap: 'wrap' as 'wrap',
    }
})

export interface Props extends WithStyles<typeof styles> {
}

export interface State {
    fields: Field[]
}

export class Main extends React.Component<Props, State> {
    public constructor(props: Props) {
        super(props)

        this.state = {
            fields: [],
        }
    }

    private onFieldsCreate = (field: Field) => {
        const currentFields = this.state.fields
        const newField = {
            ...field,
            id: currentFields.length
                ? currentFields[currentFields.length - 1].id! + 1
                : 1
        }
        currentFields.push(newField)

        this.setState({ fields: currentFields })
    }

    private onFieldDelete = (id: number) => {
        const currentFields = this.state.fields
        const newFields = currentFields.filter(f => f.id !== id)

        this.setState({ fields: newFields })
    }


    public render() {
        const { classes } = this.props
        const { fields } = this.state

        return (
            <div className={classes.container}>
                <FormBuilder onFieldsCreate={this.onFieldsCreate} />
                <FormRenderer onDelete={this.onFieldDelete} fields={fields} />
            </div>
        )
    }
}

export default withStyles(styles)(Main)
