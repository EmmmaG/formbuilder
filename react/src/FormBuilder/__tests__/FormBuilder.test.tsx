import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import FormBuilder, { Props } from '../FormBuilder'

const props: Props = {
    onFieldsCreate: jest.fn(),
	classes: idObj,
}

describe('FormBuilder', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<FormBuilder {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
	})
})
