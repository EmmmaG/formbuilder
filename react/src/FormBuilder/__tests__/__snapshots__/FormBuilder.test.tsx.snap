// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBuilder renders correctly 1`] = `
<div>
  <WithStyles(Typography)>
    Please select a field type to create
  </WithStyles(Typography)>
  <WithStyles(FormControl)
    variant="outlined"
  >
    <WithStyles(WithFormControlContext(Select))
      className="FormBuilder-select-1"
      classes={
        Object {
          "root": "FormBuilder-selectRoot-2",
        }
      }
      input={
        <WithStyles(OutlinedInput)
          id="type"
          labelWidth={0}
          name="field type"
        />
      }
      native={true}
      onChange={[Function]}
      style={
        Object {
          "width": "250px",
        }
      }
      value="textInput"
    >
      <option
        key="textInput"
        value="textInput"
      >
        Text Input
      </option>
      <option
        key="radioSelect"
        value="radioSelect"
      >
        Radio Select
      </option>
      <option
        key="checkbox"
        value="checkbox"
      >
        Checkbox
      </option>
    </WithStyles(WithFormControlContext(Select))>
  </WithStyles(FormControl)>
  <div>
    <WithStyles(TextInputBuilder)
      onCreate={[MockFunction]}
    />
  </div>
</div>
`;
